lib/libc/gen/fdopendir.c

```diff
@@ -48,8 +48,14 @@
 DIR *
 fdopendir(int fd)
 {
+	int flags;
 
-	if (_fcntl(fd, F_SETFD, FD_CLOEXEC) == -1)
+	flags = _fcntl(fd, F_GETFD, 0);
+	if (flags == -1)
+		return (NULL);
+
+	if ((flags & FD_CLOEXEC) == 0 &&
+	    _fcntl(fd, F_SETFD, flags | FD_CLOEXEC) == -1)
```

Comment:
Please move actions out of the if () conditions.  I.e. something like
`  if ((flags & FD_CLOEXEC) == 0) {
      res = _fcntl(fd, F_SETFD, flags | FD_CLOEXEC);
      if (res == -1)
          return (NULL);
   }
   return (__opendir_common(...));
`
and so on for all places you change.

Reply:
Fixed.